views/dashboard.ejs

```diff
@@ -9,74 +9,116 @@
+            </tr>
+        </thead>
+        <tbody>
+            <% if (data.length > 0) { %>
+                <% data.forEach(item => { %>
                     <tr>
-                        <td><%= script %></td>
-                        <td><%= scriptCounts[script] %></td>
+                        <td><%= item.endpoint %></td>
+                        <td><%= item.script %></td>
+                        <td><%= item.execution_time %></td>
+                        <td class="status-<%= item.status %>"><%= item.status %></td>
                     </tr>
-                <% } %>
-            </tbody>
-        </table>
-    </div>
+                <% }) %>
+            <% } else { %>
+                <tr>
+                    <td colspan="4">No data available</td>
+                </tr>
+            <% } %>
+        </tbody>
+    </table>
+
+    <script>
+        document.addEventListener('DOMContentLoaded', function() {
+            const data = <%- JSON.stringify(data) %>;
+    
+            // Summary of Script Execution Statuses Data
+            const statusCounts = { 'N/A': 0, 'error': 0, 'ran': 0 };
+            data.forEach(item => {
+                statusCounts[item.status] = (statusCounts[item.status] || 0) + 1;
+            });
+    
+            const statusData = {
+                labels: ['N/A', 'Error', 'Ran'],
+                datasets: [{
+                    data: [statusCounts['N/A'], statusCounts['error'], statusCounts['ran']],
+                    backgroundColor: ['#FFCE56', '#FF6384', '#36A2EB'],
+                }]
+            };
+    
+            // Render the Status Chart
+            const statusCtx = document.getElementById('status-chart').getContext('2d');
+            const statusChart = new Chart(statusCtx, {
+                type: 'pie',
+                data: statusData,
+            });
+    
+    
+            // Most Frequently Executed Scripts Data
+            const scriptCounts = {};
+            data.forEach(item => {
+                scriptCounts[item.script] = (scriptCounts[item.script] || 0) + 1;
+            });
+    
+            const scriptLabels = Object.keys(scriptCounts);
+            const scriptDataValues = Object.values(scriptCounts);
+    
+            const scriptData = {
+                labels: scriptLabels,
+                datasets: [{
+                    data: scriptDataValues,
+                    backgroundColor: scriptLabels.map((label, index) => {
+                        const colors = ['#FFCE56', '#FF6384', '#36A2EB', '#4BC0C0', '#9966FF', '#FF9F40'];
+                        return colors[index % colors.length];
+                    })
+                }]
+            };
+    
+            // Render the Script Chart
+            const scriptCtx = document.getElementById('script-chart').getContext('2d');
+            new Chart(scriptCtx, {
+                type: 'pie',
+                data: scriptData,
+            });
+        });
+    </script>
```

Comment:
Create a file in `public/js` and call it `dashboard.js`, and insert a link in the html to load it.
It is hard to manage scripts inside html or ejs files

Reply:
Done